rocketpy/Flight.py

```diff
@@ -2164,35 +2164,46 @@ def acceleration(self):
     @cached_property
     def max_acceleration_power_on(self):
         """Maximum acceleration reached by the rocket during motor burn."""
-        closest_value = float("inf")
-        index_to_slice = None
-
-        for index, item in enumerate(self.acceleration):
-            if abs(item[0] - self.rocket.motor.burn_out_time) < abs(
-                closest_value - self.rocket.motor.burn_out_time
-            ):
-                closest_value = item[0]
-                index_to_slice = index
+        burn_out_time_index = find_closest(
+            self.acceleration.source[:, 0], self.rocket.motor.burn_out_time
+        )
         max_acceleration_time_index = np.argmax(
-            self.acceleration[: index_to_slice + 1, 1]
+            self.acceleration[:burn_out_time_index, 1]
         )
         return self.acceleration[max_acceleration_time_index, 1]
 
     @cached_property
     def max_acceleration_power_on_time(self):
         """Time at which the rocket reaches its maximum acceleration during
         motor burn."""
-        closest_value = float("inf")
-        index_to_slice = None
-
-        for index, item in enumerate(self.acceleration):
-            if abs(item[0] - self.rocket.motor.burn_out_time) < abs(
-                closest_value - self.rocket.motor.burn_out_time
-            ):
-                closest_value = item[0]
-                index_to_slice = index
+        burn_out_time_index = find_closest(
+            self.acceleration.source[:, 0], self.rocket.motor.burn_out_time
+        )
+        max_acceleration_time_index = np.argmax(
+            self.acceleration[:burn_out_time_index, 1]
+        )
+        return self.acceleration[max_acceleration_time_index, 0]
+
+    @cached_property
+    def max_acceleration_power_off(self):
+        """Maximum acceleration reached by the rocket after motor burn."""
+        burn_out_time_index = find_closest(
+            self.acceleration.source[:, 0], self.rocket.motor.burn_out_time
+        )
+        max_acceleration_time_index = np.argmax(
+            self.acceleration[burn_out_time_index:, 1]
+        )
+        return self.acceleration[max_acceleration_time_index, 1]
+
+    @cached_property
+    def max_acceleration_power_off_time(self):
+        """Time at which the rocket reaches its maximum acceleration after
+        motor burn."""
+        burn_out_time_index = find_closest(
+            self.acceleration.source[:, 0], self.rocket.motor.burn_out_time
+        )
         max_acceleration_time_index = np.argmax(
-            self.acceleration[: index_to_slice + 1, 1]
+            self.acceleration[burn_out_time_index:, 1]
         )
         return self.acceleration[max_acceleration_time_index, 0]
```

Comment:
These two methods are essentially the same. Personally, I would keep the `max_acceleration_power_off_time` as it is, and use its value in the `max_acceleration_power_off` as

```py
return self.acceleration(self.max_acceleration_power_off_time)
```

Which is similar to the way that the methods `max_speed` and `max_speed_time` are implemented.